cowprotocol/cowswap
# (feat: hook-store): create bundle hooks tenderly simulation

# Summary

Add bundle simulations on the hook store feature. This enhances the hooks simulations by considering all the previous hooks and the swap (in case of post hooks). 

https://www.loom.com/share/37cf070b4f7b41f896966c469941c406?sid=c183f6f3-f611-4d2f-8d7d-0015bb8d374a

# To Test

 Since we still don't have a Tenderly account, it is expected to have to login on Tenderly to see the simulation links.

1. Run the bff locally on port 3001. For more information check: https://github.com/cowprotocol/bff/pull/88 
2. Change the bff connection by editing the file: `libs/common-const/src/bff.ts`. Change the default URL to `http://localhost:3001`
3. Run the cowswap app locally.

4. Go to the hooks tab of the swap widget and add some hooks. 
- [ ] Should be able to see Tenderly links for failing transactions.
- [ ] Should auto refresh simulations on hook or token changes.
- [ ] Check the simulation for different tokens to sell and buy.
- [ ] Should be able to create order even if hooks are falling.

# Background

@alfetopito @shoom3301 In order to merge this PR we will have to add the .env keys. Let me know if you think that we should wait for the refactoring to the backend for the frontend repo.
